hudi-client/src/main/java/org/apache/hudi/table/action/rollback/CopyOnWriteRollbackActionExecutor.java

```diff
@@ -0,0 +1,94 @@
+
+  public CopyOnWriteRollbackActionExecutor(JavaSparkContext jsc,
+                                           HoodieWriteConfig config,
+                                           HoodieTable<?> table,
+                                           String instantTime,
+                                           HoodieInstant commitInstant,
+                                           boolean deleteInstants) {
+    super(jsc, config, table, instantTime, commitInstant, deleteInstants);
+  }
+
+  public CopyOnWriteRollbackActionExecutor(JavaSparkContext jsc,
+                                           HoodieWriteConfig config,
+                                           HoodieTable<?> table,
+                                           String instantTime,
+                                           HoodieInstant commitInstant,
+                                           boolean deleteInstants,
+                                           boolean skipTimelinePublish) {
+    super(jsc, config, table, instantTime, commitInstant, deleteInstants, skipTimelinePublish);
+  }
+
+  @Override
+  protected List<HoodieRollbackStat> executeRollback() throws IOException {
+    long startTime = System.currentTimeMillis();
+    List<HoodieRollbackStat> stats = new ArrayList<>();
+    HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
+    HoodieInstant resolvedInstant = instantToRollback;
+
+    if (instantToRollback.isCompleted()) {
+      LOG.info("Unpublishing instant " + instantToRollback);
+      resolvedInstant = activeTimeline.revertToInflight(instantToRollback);
+    }
+
+    // For Requested State (like failure during index lookup), there is nothing to do rollback other than
+    // deleting the timeline file
+    if (!resolvedInstant.isRequested()) {
+      // delete all the data files for this commit
+      LOG.info("Clean out all parquet files generated for commit: " + resolvedInstant);
+      List<RollbackRequest> rollbackRequests = generateRollbackRequests(resolvedInstant);
+
+      //TODO: We need to persist this as rollback workload and use it in case of partial failures
+      stats = new RollbackHelper(table.getMetaClient(), config).performRollback(jsc, resolvedInstant, rollbackRequests);
+    }
+    // Delete Inflight instant if enabled
+    deleteInflightAndRequestedInstant(deleteInstants, activeTimeline, resolvedInstant);
+    LOG.info("Time(in ms) taken to finish rollback " + (System.currentTimeMillis() - startTime));
+    return stats;
+  }
+
+  private List<RollbackRequest> generateRollbackRequests(HoodieInstant instantToRollback)
+      throws IOException {
+    return FSUtils.getAllPartitionPaths(table.getMetaClient().getFs(), table.getMetaClient().getBasePath(),
+        config.shouldAssumeDatePartitioning()).stream()
+        .map(partitionPath -> RollbackRequest.createRollbackRequestWithDeleteDataAndLogFilesAction(partitionPath, instantToRollback))
```

Comment:
This is not related to current change. Wondering why COW rollback needs to deal with  log files rollback. Should not be a correctness issue but would warrant a fix. I can open a jira and look at this later if you also think this is weird.

Reply:
it is.. https://github.com/apache/incubator-hudi/blob/master/hudi-client/src/main/java/org/apache/hudi/table/HoodieCopyOnWriteTable.java#L237 does call this.. and this is the commit that introduce this.. https://github.com/apache/incubator-hudi/pull/968/files#diff-47ab1d64d6b64228b07d3c4c4559786cR328

def worth following up